cpp/foxglove/include/foxglove/channel.hpp

```diff
@@ -35,6 +35,55 @@ struct Schema {
   size_t data_len = 0;
 };
```

Comment:
This looks good, but there seems to be a lifetime issue with the callback context (maybe applies to some of our other C++ callbacks too?)

Doesn't seem to be a way to free it. What do you think?